idpi/src/operators/vertical_interpolation.py

```diff
@@ -0,0 +1,363 @@
+
+    # ATTENTION: the attribute "positive" is not set for generalVerticalLayer
+    #            we know that for COSMO it would be defined as positive:"down"; for the time being,
+    #            we explicitly use the height field on model mid layer surfaces as auxiliary field
+
+    # Parameters
+    # ... supported folding modes
+    folding_modes = ("low_fold", "high_fold", "undef_fold")
+    if mode not in folding_modes:
+        raise RuntimeError("interpolate_k2theta: unsupported mode", mode)
+
+    # ... supported tc units and corresponding conversion factor to K (i.e. to the same unit as theta); according to GRIB2
+    #     isentropic surfaces are coded in K; fieldextra codes them in cK for NetCDF (to be checked)
+    tcth_unit_conversions = dict(K=1.0, cK=0.01)
+    if tcth_units not in tcth_unit_conversions.keys():
+        raise RuntimeError(
+            "interpolate_k2theta: unsupported value of tcth_units", tcth_units
+        )
+    # ... supported range of tc values (in K)
+    tc_min = 1.0
+    tc_max = 1000.0
+    # ... tc values outside range of meaningful values of height, used in tc interval search (in m amsl)
+    h_min = -1000.0
+    h_max = 100000.0
+
+    # Define vertical target coordinates
+    tc = dict()
+    tc_values = tcth_values.copy()
+    tc_values.sort(
+        reverse=False
+    )  # Sorting cannot be exploited for optimizations, since theta is not monotonous wrt to height
+    # tc values are stored in K
+    tc["values"] = np.array(tc_values)
+    tc_factor = tcth_unit_conversions[tcth_units]
+    tc["values"] *= tc_factor
+    if min(tc["values"]) < tc_min or max(tc["values"]) > tc_max:
+        raise RuntimeError(
+            "interpolate_k2theta: target coordinate value out of range (must be in interval [",
+            tc_min,
+            ", ",
+            tc_max,
+            "]K)",
+        )
+    tc["attrs"] = {
+        "units": "K",
+        "positive": "up",
+        "standard_name": "air_potential_temperature",
+        "long_name": "potential temperature",
+    }
+    tc["typeOfLevel"] = "theta"  # not yet properly defined in eccodes
+    tc["NV"] = 0
+
+    # Prepare output field ftc on tc
+    # name
+    ftc_name = field.name
+    # attrs
+    ftc_attrs = field.attrs.copy()
+    ftc_attrs["GRIB_typeOfLevel"] = tc["typeOfLevel"]
+    if ftc_attrs["GRIB_NV"] is not None:
+        ftc_attrs["GRIB_NV"] = tc["NV"]
+    # dims
+    ftc_shape = list(
+        len(field[d]) if d != "generalVerticalLayer" else len(tc["values"])
+        for d in field.dims
+    )
+    ftc_dims = list(
+        map(lambda x: x.replace("generalVerticalLayer", tc["typeOfLevel"]), field.dims)
+    )
+    ftc_dims = tuple(ftc_dims)
+    ftc_shape = tuple(ftc_shape)
+    # coords
+    # ... inherit all except for the vertical coordinates
+    ftc_coords = dict()
+    for c in field.coords:
+        if c != "generalVerticalLayer":
+            ftc_coords[c] = field.coords[c]
```

Comment:
same one liner as above using for loop comprehesion: 
```ftc_coords = {c:v for c,v in field.coords.items() if c != 'generalVerticalLayer'}```

Reply:
Done.